src/Contracts/Container/Container.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+/**
+ * This file is part of the Zest Framework.
+ *
+ * @author   Muhammad Umer Farooq <lablnet01@gmail.com>
+ * @author-profile https://www.facebook.com/Muhammadumerfarooq01/
+ *
+ * For the full copyright and license information, please view the LICENSE
+ *  file that was distributed with this source code.
+ *
+ * @since 3.0.0
+ *
+ * @license MIT
+ */
+
+namespace Zest\Contracts\Container;
+
+interface Container
+{
+    /**
+     * Register a type hint.
+     *
+     * @param string|array    $hint      Type hint or array contaning both type hint and alias.
+     * @param string|\Closure $class     Class name or closure.
+     * @param bool            $singleton Should we return the same instance every time?
+     *
+     * @since 3.0.0
+     *
+     * @return void
+     */
+    public function register($hint, $class, bool $singleton = false);
+
+    /**
+     * Register a type hint and return the same instance every time.
+     *
+     * @param string|array    $hint  Type hint or array contaning both type hint and alias.
+     * @param string|\Closure $class Class name or closure.
+     *
+     * @since 3.0.0
+     *
+     * @return 3.0.0
+     */
+    public function registerSingleton($hint, $class);
+
+    /**
+     * Register a singleton instance.
+     *
+     * @param string|array $hint     Type hint or array contaning both type hint and alias.
+     * @param object       $instance Class instance.
+     *
+     * @since 3.0.0
+     *
+     * @return void
+     */
+    public function registerInstance($hint, $instance);
+
+    /**
+     * Registers a contextual dependency.
+     *
+     * @param string $class          Class.
+     * @param string $interface      Interface.
+     * @param string $implementation Implementation.
+     *
+     * @since 3.0.0
+     *
+     * @return void
+     */
+    public function registerContextualDependency($class, string $interface, string $implementation);
```

Comment:
I think it should add the `sting` type hint for `$class`.

Reply:
explain more please

Reply:
It can use the `class_exists` to validate `$class`.